### PR TITLE
Show error about wrong time on sync init

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
## Submitter Checklist:

Fixes https://github.com/brave/brave-browser/issues/3962.

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues/3962) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [ ] macOS
  - [x] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
1. Adjust computer time 20 minutes forward.
2. Create a sync chain
3. Should see:`You might be having wrong system time or timezone setup on your device. Please setup the correct time and timezone`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by AlexeyBarabash** [comment]
Linked brave-sync PR https://github.com/brave/sync/pull/290 .

**Comment by AlexeyBarabash** [comment]
CI build succeeded on Windows and Linux, and failed on MacOS with an error of applying patches.

File: components/brave_sync/extension/background.js [hunk 1]
```diff
@@ -234,8 +234,10 @@ class InjectedObject {
         chrome.braveSync.getInitData(arg1/*syncVersion*/);
         break;
       case "sync-setup-error":
-        console.log(`"sync-setup-error" error=${JSON.stringify(arg1)}`);
-        chrome.braveSync.syncSetupError('ERR_SYNC_INIT_FAILED');
+        console.log(`"sync-setup-error" error=${arg1}`);
+        var errorToPass = (arg1 === 'Credential server response 400. Signed request body of the client timestamp is required.') ?
```
Comment (cezaraugusto) [inline components/brave_sync/extension/background.js]: after changing my time I get the `Credential server response 400` error but not the message after that so it fails under `ERR_SYNC_INIT_FAILED` to me

**Comment by bbondy** [comment]
Hey guys, for text changes let's please include someone from #product for review. Thanks!

**Comment by bbondy** [comment]
This will need some adjusting please, please check with the product team to get the wording.

**Comment by kjozwiak** [comment]
Created https://github.com/brave/brave-browser/issues/4011 to clean up the text 👍

**Pull request merged**